server/src/main/java/com/hedera/block/server/verification/service/BlockVerificationServiceImpl.java

```
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.verification.service;
+
+import static java.lang.System.Logger.Level.WARNING;
+
+import com.hedera.block.server.metrics.BlockNodeMetricTypes;
+import com.hedera.block.server.metrics.MetricsService;
+import com.hedera.block.server.verification.session.BlockVerificationSession;
+import com.hedera.block.server.verification.session.BlockVerificationSessionFactory;
+import com.hedera.hapi.block.BlockItemUnparsed;
+import com.hedera.hapi.block.stream.output.BlockHeader;
+import com.hedera.pbj.runtime.ParseException;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import javax.inject.Inject;
+
+/**
+ * Service that handles the verification of block items, it receives items from the handler.
+ */
+public class BlockVerificationServiceImpl implements BlockVerificationService {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+    private final MetricsService metricsService;
+
+    private final BlockVerificationSessionFactory sessionFactory;
+    private BlockVerificationSession currentSession;
+
+    /**
+     * Constructs a new BlockVerificationServiceImpl.
+     *
+     * @param metricsService the metrics service
+     * @param sessionFactory the session factory
+     */
+    @Inject
+    public BlockVerificationServiceImpl(
+            @NonNull final MetricsService metricsService,
+            @NonNull final BlockVerificationSessionFactory sessionFactory) {
+        this.metricsService = metricsService;
+        this.sessionFactory = sessionFactory;
```

add require non null checks on 53-54 as they are assignment only, lets fail early here.

---

✅ Done!

---

It seems the change was reverted?! (I believe unintentionally)

---

I also forgot to do this one, I most have mistakenly for the other similar comments.